transmart-copy/src/test/groovy/org/transmartproject/copy/CopySpec.groovy

```diff
@@ -162,11 +135,107 @@ class CopySpec extends Specification {
         ] as Set
     }
 
-    Map<Table, Number> count(Database database, Iterable<Table> tables) {
+    def 'delete by path when no study'() {
+        given: 'there is no study'
+        ensureTestStudyUnloaded()
+
+        when:
+        copy.deleteStudy(STUDY_FOLDER, false)
+        then:
+        !getTestStudyDbIdentifier()
+
+        when:
+        copy.deleteStudy(STUDY_FOLDER, true)
+        then:
+        def exception = thrown(IllegalStateException)
+        exception.message == 'Study not found: ' + TEST_STUDY + '.'
+    }
+
+    def 'delete by path when study loaded'() {
+        given: 'there is the study'
+        ensureTestStudyLoaded()
+
+        when:
+        copy.deleteStudy(STUDY_FOLDER)
+        then:
+        !getTestStudyDbIdentifier()
+    }
+
+    def 'delete exclude default upload'() {
+        given: 'there is the study'
+        ensureTestStudyLoaded()
+        Options options = new Options()
+        options.addOption(new Option('d', 'delete', true, ''))
+
+        when: 'error during load happens'
+        CommandLine cli1 = new DefaultParser()
+                .parse(options, ['--delete', TEST_STUDY] as String[])
+        Copy.runCopy(cli1, DATABASE_CREDENTIALS)
+        then: 'study has been deleted'
+        !getTestStudyDbIdentifier()
```

Comment:
The type of `getTestStudyDbIdentifier` is `Number`. If the study has id `0`, this fails?